packages/ember-metal/lib/watch_key.js

```diff
@@ -16,8 +16,6 @@ import {
 let handleMandatorySetter;
 
 export function watchKey(obj, keyName, _meta) {
-  if (typeof obj !== 'object' || obj === null) { return; }
```

Comment:
since `meta` has assertion for non-object / non-function values https://github.com/emberjs/ember.js/blob/76dc4deb70cd1e5ae7d83374161925af50946a01/packages/ember-metal/lib/meta.js#L554 
so it is better to assert than be silent to uncover unnecessary calls of `watchKey `

Reply:
does the assert check for null objects?

Reply:
yes https://github.com/emberjs/ember.js/blob/76dc4deb70cd1e5ae7d83374161925af50946a01/packages/ember-metal/lib/meta.js#L554